### DISCUSSION TITLE
Service account

### DISCUSSION THREAD

**Discussion by tohodakilla** [pocketbase/pocketbase#1450] [Q&A] [answered]
What kind of API rules I can add to allow some kind of "service account" access - no login only very long lived token.
Use case: i want to monitor progress of multiple videos transcoding process. For that im thinking send webhook that can access above mentioned "service account" and update records in pocketbase collection.

**Comment by ganigeorgiev** [comment, marked as answer]
You can't control the tokens duration via an API rule, but you can adjust them globally from "Admin UI > Settings > Token options" (by default PocketBase auth tokens are usually long lived, eg. 14 days).

Permanent admin tokens (or API keys) currently are not supported but it is in my todo to consider it and eventually may be added in the near future since several users already requested it (you can also check https://github.com/pocketbase/pocketbase/discussions/342).

**Reply by tohodakilla** [reply]
thanks for prompt response. and what is this `@request.data.*` ?
maybe i can send some string through that?
![Screenshot from 2022-12-31 16-56-27](https://user-images.githubusercontent.com/1858683/210148932-56755bd6-7f0e-44af-b2ce-0b42a6445cd5.png)

**Reply by ganigeorgiev** [reply]
`@request.data.*` is the body data that was sent with your request.
Similarly, `@request.query.*` is used for accessing the request query parameters.
More information you could find in https://pocketbase.io/docs/api-rules-and-filters/#filters-syntax.

**Reply by tohodakilla** [reply]
so it seems my use case can be satisfied with @request.data.token = 'my super unique long lived secret token'

all the best :heart: :pray:

**Comment by tohodakilla** [comment]
Yes i've solved my use case with API rule:
`(...) || @request.query.secretKey = "mySuperLongAndSecretKeyThatNooneCanGuess"`
`@request.data` works too, but I had to use custom solution to add data to GET requests, hence I've chosen `@request.query`
